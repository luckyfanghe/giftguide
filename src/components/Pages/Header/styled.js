import styled from "styled-components";
import { mainWidth } from "../../../constants/utils";

export const Body = styled.div`
  width: ${mainWidth}%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: auto;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Topbar = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Logo = styled.div`
  display: flex;
  align-items: center;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const LogoText = styled.span`
  padding-left: 15px;
  font-size: 30px;
  vertical-align: middle;
  text-align: center;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Avatar = styled.div`
  display: flex;
  align-items: center;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const SearchInput = styled.div`
  display: ${props => (props.visible == "false" ? "none" : "block")};
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const Plus = styled.span`
  padding-left: 10px;
  padding-right: 10px;
  font-size: 30px;
  font-weight: bold;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

export const AvatarImg = styled.img.attrs(props => ({
  alt: props.alt,
  src: props.src
}))`
  width: 30px;
  height: 30px;
  border-radius: 1rem;
  border: ${props => (props.border ? 1 : 0)}px solid black;
`;

// export const NabVar = styled.Navbar`
//   padding: 0px;
//   border: ${props => (props.border ? 1 : 0)}px solid black;
// `;
