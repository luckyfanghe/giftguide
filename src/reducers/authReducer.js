// @flow
import types from "../actions/types";

const initialState = {
  sidebarOpened: true,
  sidebarStyle: "login",
  authState: ""
};
function authReducer(state = initialState, action) {
  switch (action.type) {
    case types.AUTH.OPENSIDEBAR:
      return {
        ...state,
        sidebarOpened: true,
        sidebarStyle: action.sidebarStyle
      };
    case types.AUTH.CLOSESIDEBAR:
      return {
        ...state,
        sidebarOpened: false,
        sidebarStyle: action.sidebarStyle
      };
    case types.AUTH.DOINGAUTH:
      return {
        ...state,
        authState: action.state
      };
    default:
      return {
        ...state
      };
  }
}

export default authReducer;
