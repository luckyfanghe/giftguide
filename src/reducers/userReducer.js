// @flow
import types from "../actions/types";

const initialState = {
  user: null,
  userState: "",
  userStateDes: ""
};
function userReducer(state = initialState, action) {
  switch (action.type) {
    case types.USER.UPDATE:
      return {
        ...state,
        user: action.userInfo
      };
    case types.USER.LOGINFAILED:
      return {
        ...state,
        userState: "login failed",
        userStateDes: action.state
      };
    case types.USER.LOGINSUCCESS:
      return {
        ...state,
        userState: "login success",
        userStateDes: ""
      };
    case types.USER.SIGNUPFAILED:
      return {
        ...state,
        userState: "signup failed",
        userStateDes: action.state
      };
    case types.USER.SIGNUPSUCCESS:
      return {
        ...state,
        userState: "signup success",
        userStateDes: ""
      };
    default:
      return {
        ...state
      };
  }
}

export default userReducer;
