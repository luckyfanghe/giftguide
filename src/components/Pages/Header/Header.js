import React from "react";
import { bindActionCreators } from "redux";
import { connect } from "react-redux";
import { Link, withRouter } from "react-router-dom";
import { Navbar, Nav, FormControl } from "react-bootstrap";

import * as S from "./styled";
import logo from "../../../assets/images/Logo.webp";
import searchIcon from "../../../assets/images/search-icon.webp";
import avatarImg from "../../../assets/images/avatar.webp";
import * as routes from "../../../constants/routes";

import "../../Styles/style.css";

class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      isSearchInputVisible: false
    };
  }

  _toggleSearch = e => {
    e.preventDefault();
    this.setState({
      isSearchInputVisible: this.state.isSearchInputVisible ? false : true
    });
  };

  _handleKeyDown = e => {
    if (e.key === "Enter") {
      this.props.history.push(routes.SEARCH_PAGE);
    }
  };

  render() {
    return (
      <React.Fragment>
        <S.Body data-testid="Header">
          <S.Topbar>
            <S.Logo>
              <img src={logo} alt="logo" />
              <S.LogoText>THE GIFT GUIDE</S.LogoText>
            </S.Logo>
            <S.Avatar>
              <S.SearchInput
                visible={this.state.isSearchInputVisible ? "ture" : "false"}
              >
                <FormControl
                  type="text"
                  placeholder="Search"
                  onKeyDown={this._handleKeyDown}
                />
              </S.SearchInput>
              <img src={searchIcon} alt="search" onClick={this._toggleSearch} />
              <S.Plus>+</S.Plus>
              <S.AvatarImg src={avatarImg} alt="avatar" />
            </S.Avatar>
          </S.Topbar>
          <div className="nab-bar">
            <Navbar bg="light" expand="lg" className="nab-bar-bar">
              <Navbar.Toggle aria-controls="basic-navbar-nav" />
              <Navbar.Collapse id="basic-navbar-nav" className="nab-bar-link">
                <Nav className="mr-auto">
                  <Nav.Link to="/dashboard" className="nab-link" as={Link}>
                    DASHBOARD
                  </Nav.Link>
                  <Nav.Link to="/wish_list" className="nab-link" as={Link}>
                    WISH LIST
                  </Nav.Link>
                  <Nav.Link to="/friends" className="nab-link" as={Link}>
                    FRIENDS
                  </Nav.Link>
                  <Nav.Link to="/calendar" className="nab-link" as={Link}>
                    CALENDAR
                  </Nav.Link>
                  <Nav.Link to="/gift_guides" className="nab-link" as={Link}>
                    GIFT GUIDES
                  </Nav.Link>
                  <Nav.Link to="/blog" className="nab-link" as={Link}>
                    BLOG
                  </Nav.Link>
                  <Nav.Link to="/contact" className="nab-link" as={Link}>
                    CONTACT
                  </Nav.Link>
                  <Nav.Link to="/about" className="nab-link" as={Link}>
                    ABOUT
                  </Nav.Link>
                </Nav>
              </Navbar.Collapse>
            </Navbar>
          </div>
        </S.Body>
      </React.Fragment>
    );
  }
}

const actions = dispatch => bindActionCreators({}, dispatch);

function mapStateToProps(state) {
  return {
    user: state.userReducer.user,
    userState: state.userReducer.userState
  };
}

// export default Header;
export default connect(mapStateToProps, actions)(withRouter(Header));
