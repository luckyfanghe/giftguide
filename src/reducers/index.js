import { combineReducers } from "redux";
import { connectRouter } from "connected-react-router";

import userReducer from "./userReducer";
import authReducer from "./authReducer";

export default combineReducers({
  userReducer,
  authReducer
});
