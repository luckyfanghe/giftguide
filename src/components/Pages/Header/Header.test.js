import React from "react";
import { render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import Header from "./Header";

import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { createStore, applyMiddleware, compose } from "redux";
import createSagaMiddleware from "redux-saga";
import reducers from "../../../reducers";
import rootSaga from "../../../sagas";

const sagaMiddleware = createSagaMiddleware();
const enhancers = composeWithDevTools(applyMiddleware(sagaMiddleware));

const store = createStore(reducers, enhancers);

sagaMiddleware.run(rootSaga);

it("renders corrently", () => {
  const { queryByTestId } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    </Provider>
  );

  expect(queryByTestId("Header")).toBeTruthy();
});

describe("Header", () => {
  test("Header snapshot renders ", () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
