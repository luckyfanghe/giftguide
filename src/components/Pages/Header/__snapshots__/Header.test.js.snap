// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header snapshot renders  1`] = `
<div
  className="sc-AykKC jIyKXn"
  data-testid="Header"
>
  <div
    className="sc-AykKD PnOJm"
  >
    <div
      className="sc-AykKE dtIxTI"
    >
      <img
        alt="logo"
        src="Logo.webp"
      />
      <span
        className="sc-AykKF eknJnN"
      >
        THE GIFT GUIDE
      </span>
    </div>
    <div
      className="sc-AykKG iQLzuy"
    >
      <div
        className="sc-AykKH gThAOh"
      >
        <input
          className="form-control"
          onKeyDown={[Function]}
          placeholder="Search"
          type="text"
        />
      </div>
      <img
        alt="search"
        onClick={[Function]}
        src="search-icon.webp"
      />
      <span
        className="sc-AykKI fiqlbo"
      >
        +
      </span>
      <img
        alt="avatar"
        className="sc-AykKJ eAebmr"
        src="avatar.webp"
      />
    </div>
  </div>
  <div
    className="nab-bar"
  >
    <nav
      className="nab-bar-bar navbar navbar-expand-lg navbar-light bg-light"
    >
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        className="navbar-toggler collapsed"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        aria-expanded={null}
        className="nab-bar-link navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          className="mr-auto navbar-nav"
          onKeyDown={[Function]}
        >
          <a
            className="nab-link nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/dashboard"
            onClick={[Function]}
          >
            DASHBOARD
          </a>
          <a
            className="nab-link nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/wish_list"
            onClick={[Function]}
          >
            WISH LIST
          </a>
          <a
            className="nab-link nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/friends"
            onClick={[Function]}
          >
            FRIENDS
          </a>
          <a
            className="nab-link nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/calendar"
            onClick={[Function]}
          >
            CALENDAR
          </a>
          <a
            className="nab-link nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/gift_guides"
            onClick={[Function]}
          >
            GIFT GUIDES
          </a>
          <a
            className="nab-link nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/blog"
            onClick={[Function]}
          >
            BLOG
          </a>
          <a
            className="nab-link nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/contact"
            onClick={[Function]}
          >
            CONTACT
          </a>
          <a
            className="nab-link nav-link"
            data-rb-event-key={null}
            disabled={false}
            href="/about"
            onClick={[Function]}
          >
            ABOUT
          </a>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
